### ISSUE TITLE
No Table Found & No Default Content Found & An Error Has Occured

### ISSUE THREAD

**Issue by TheNuclearCat** [passbolt/passbolt_api#164]
```
---------------------------------------------------------------
     ____                  __          ____
    / __ \____  _____ ____/ /_  ____  / / /_
   / /_/ / __ `/ ___/ ___/ __ \/ __ \/ / __/
  / ____/ /_/ (__  |__  ) /_/ / /_/ / / /
 /_/    \__,_/____/____/_.___/\____/_/\__/

 Open source password manager for teams
---------------------------------------------------------------
 Healthcheck shell
---------------------------------------------------------------

 Environment

 [PASS] PHP version 7.0.22-0ubuntu0.16.04.1
 [PASS] PCRE compiled with unicode support
 [PASS] The app/tmp directory is writable
 [PASS] The app/webroot/img/public directory is writable

 Config files

 [PASS] The core config file is present
 [PASS] The database config file is present
 [PASS] The email config file is present
 [PASS] The application config file is present

 Core config

 [PASS] Debug mode is off.
 [PASS] Cache is working.
 [PASS] Unique value set for security.salt
 [PASS] Unique value set for security.cipherSeed
 [PASS] Full base url is set to https://passbolt.flawyn.de
 [PASS] App.fullBaseUrl validation OK.
 [PASS] /healthcheck/status is reachable.

 SSL Certificate

 [PASS] SSL peer certificate validates
 [PASS] Hostname is matching in SSL certificate.
 [PASS] Not using a self-signed certificate

 Database

 [PASS] Configured to use a supported database backend
 [PASS] The application is able to connect to the database
 [PASS] Not using a prefix for database tables
 [FAIL] No table found
  [HELP] Run the install script to install the database tables
  [HELP] su -s /bin/bash -c "./app/Console/cake install" www-data
 [FAIL] No default content found
  [HELP] Run the install script to set the default content such as roles and permission types
 [PASS] The database schema up to date.

 GPG Configuration

 [PASS] PHP GPG Module is installed and loaded
 [PASS] The server gpg key is not the default one
 [PASS] The environment variable GNUPGHOME is set to /var/www/vhosts/flawyn.de/.gnupg
 [PASS] The directory containing the keyring is writable by flawyn.
 [PASS] The public key file is defined in app/config.php and readable.
 [PASS] The public key file is defined in app/config.php and readable.
 [PASS] The server key fingerprint matches the one defined in app/config.php.
 [PASS] There is a valid email id defined for the server key.

 Application configuration

 [PASS] Using latest passbolt version (1.6.4)
 [PASS] Passbolt is configured to force SSL use
 [PASS] App.fullBaseUrl is set to HTTPS
 [PASS] Selenium API endpoints are disabled.
 [PASS] Search engine robots are told not to index content.
 [PASS] Registration is closed, only administrators can add users.
 [PASS] Serving the compiled version of the javascript app
 [PASS] All email notifications will be sent.

 Development Tools (optional)

 [PASS] Phpunit is installed
 [PASS] Phpunit version is 3.7.38

  2 error(s) found. Hang in there!
```

The Installer Script runs without any errors and the database has the data in it. But If I run the health check it is complaining about errors and the website only says  "An Internal Error Has Occurred."

**Comment by cedricalfonsi** [comment]
Could you please copy/paste the cakephp log file ? it could give use more information.
`app/tmp/logs/error.log`

**Comment by TheNuclearCat** [comment]
```
#3 /var/www/vhosts/domain.de/passbolt.domain.de/app/Controller/AuthController.php(35): AuthComponent->login()
#4 [internal function]: AuthController->login()
#5 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Controller/Controller.php(491): ReflectionMethod->invokeArgs(Object(AuthController), Array)
#6 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#7 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(AuthController), Object(CakeRequest))
#8 /var/www/vhosts/domain.de/passbolt.domain.de/app/webroot/index.php(110): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#9 {main}
2017-09-02 18:51:15 Error: [Error] Class 'gnupg' not found
Request URL: /auth/login
Stack Trace:
#0 /var/www/vhosts/domain.de/passbolt.domain.de/app/Controller/Component/Auth/GpgAuthenticate.php(49): GpgAuthenticate->_initKeyring()
#1 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Controller/Component/AuthComponent.php(770): GpgAuthenticate->authenticate(Object(CakeRequest), Object(CakeResponse))
#2 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Controller/Component/AuthComponent.php(611): AuthComponent->identify(Object(CakeRequest), Object(CakeResponse))
#3 /var/www/vhosts/domain.de/passbolt.domain.de/app/Controller/AuthController.php(35): AuthComponent->login()
#4 [internal function]: AuthController->login()
#5 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Controller/Controller.php(491): ReflectionMethod->invokeArgs(Object(AuthController), Array)
#6 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#7 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(AuthController), Object(CakeRequest))
#8 /var/www/vhosts/domain.de/passbolt.domain.de/app/webroot/index.php(110): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#9 {main}
2017-09-02 20:44:47 Error: [Error] Class 'gnupg' not found
Request URL: /auth/login
Stack Trace:
#0 /var/www/vhosts/domain.de/passbolt.domain.de/app/Controller/Component/Auth/GpgAuthenticate.php(49): GpgAuthenticate->_initKeyring()
#1 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Controller/Component/AuthComponent.php(770): GpgAuthenticate->authenticate(Object(CakeRequest), Object(CakeResponse))
#2 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Controller/Component/AuthComponent.php(611): AuthComponent->identify(Object(CakeRequest), Object(CakeResponse))
#3 /var/www/vhosts/domain.de/passbolt.domain.de/app/Controller/AuthController.php(35): AuthComponent->login()
#4 [internal function]: AuthController->login()
#5 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Controller/Controller.php(491): ReflectionMethod->invokeArgs(Object(AuthController), Array)
#6 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#7 /var/www/vhosts/domain.de/passbolt.domain.de/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(AuthController), Object(CakeRequest))
#8 /var/www/vhosts/domain.de/passbolt.domain.de/app/webroot/index.php(110): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#9 {main}
```

**Comment by cedricalfonsi** [comment]
The healtcheck error and the internal error could be because a cake cache issue, your should clean the cake cache and retry.
`rm app/tmp/cache/persistent/* app/tmp/cache/models/*`

Corrupted cake cache issue can happens when the passbolt command (healthcheck/update/install) are executed as root. All the commands should be executed as http user (www-data).

**Comment by cedricalfonsi** [comment]
Fixed with v1.6.5.

**Comment by cedricalfonsi** [comment]
Hello @TheNuclearCat,

Did you check to remove the content of app/tmp/cache/persistent & app/tmp/cache/models ?

By the way, we just release a new version v1.6.5 that improves the healtcheck (gpg server key check encrypt/decrypt, app/tmp recursive permissions check).

This new version of the healthcheck could help you figure out where your issue is coming from.
